sdk/js/src/Client.ts

```diff
@@ -287,3 +295,18 @@ const checkIfEdgeEnabled = (
         return false
     }
 }
+
+const getTypeFromDefaultValue = (defaultValue: VariableValue, logger: DVCLogger) => {
+    if (typeof defaultValue === 'boolean') {
+        return VariableType.boolean
+    } else if (typeof defaultValue === 'number') {
+        return VariableType.number
+    } else if (typeof defaultValue === 'string') {
+        return VariableType.string
+    } else if (typeof defaultValue === 'object') {
+        return VariableType.object
+    } else {
+        logger.debug('Default value is not of type Boolean, Number, String, or JSON')
```

Comment:
is this log something that is meant to inform the user they're using the variable method wrong? If so, should it be a warning and include the name of the variable key being used? I imagine we might want it to log this one level higher (in the variable method) as part of the parameter validation

Reply:
I've updated it to do `logger.warn` and to include the variable key in the message. As there's a ticket for addressing the mismatch in the variable method already, is this ok?